src/core/border_router/routing_manager.cpp

```diff
@@ -817,6 +817,45 @@ bool RoutingManager::NetworkDataContainsUlaRoute(void) const
     return contains;
 }
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_REACHABLITY_CHECK_ICMP6_ERROR_ENABLE
+
+void RoutingManager::CheckReachbilityToSendIcmpError(const Message &aMessage, const Ip6::Header &aIp6Header)
+{
+    Ip6::MessageInfo messageInfo;
+
+    VerifyOrExit(IsRunning() && IsInitalPolicyEvaluationDone());
+
+    VerifyOrExit(!aIp6Header.GetDestination().IsMulticast());
+
+    // Validate that source matches the ULA OMR prefix (which is not
+    // infrastructure-derived).
+
+    VerifyOrExit(!mOmrPrefixManager.GetFavoredPrefix().IsEmpty());
+    VerifyOrExit(!mOmrPrefixManager.GetFavoredPrefix().IsInfrastructureDerived());
```

Comment:
check can be just on the "Preference" value of the ULA prefix of the source IPv6 address of the packet.
(Whether it's favored or not by the BR shouldn't play a role?)  

Is calling `IsInfrastructureDerived()` the easiest way to check the Preference value? (Haven't checked further myself - sorry)

Reply:
Yes. `IsInfrastructureDerived()` does basically check preference (and that prefix is not empty) :)

I re-wrote this whole set of checks now, so these are all removed.